libautoscoper/src/Mesh.cpp

```diff
@@ -0,0 +1,230 @@
+#include "Mesh.hpp"
+#include <fstream>
+#include <iostream>
+#include <stdexcept>
+
+Mesh::Mesh(std::string& filename){
+	//check the file extension
+	std::string extension = filename.substr(filename.find_last_of(".") + 1);
+	if (extension == "stl") {
+		ReadSTL(filename);
+	} else if (extension == "wrl" || extension == "iv") {
+		ReadVRML(filename);
+	}  else {
+		throw std::runtime_error("Mesh: File extension, " + extension + ", is not supported");
+	}
+}
+
+Mesh::~Mesh() {
+	//TODO
+}
+
+void Mesh::ReadSTL(std::string filename) {
```

Comment:
```suggestion
void Mesh::ReadSTL(const std::string& filename) {
```

Make sure the interface use const correctness where relevant